src/wifi/model/mac-low.cc

```diff
@@ -571,49 +578,76 @@ MacLow::ResumeTransmission (Time duration, Ptr<DcaTxop> dca)
     }
   else
     {
+      NS_LOG_DEBUG ("No enough time to complete this DMG transaction for Packet=" << m_currentPacket);
+      StoreAllocationParameters ();
       m_transmissionSuspended = true;
     }
-
-  /* Remove Allocation from allocation list as we restored it */
-  m_allocationPeriodsTable.erase (m_currentAllocationID);
```

Comment:
why was this moved above?

Reply:
Because we could remove the stored allocation if there's not enough time to complete the transmission of the resumed packet.